Symfony/CS/Fixer/Symfony/MethodSeparationFixer.php

```diff
@@ -0,0 +1,221 @@
+            }
+
+            $attributes = $tokensAnalyzer->getMethodAttributes($index);
+            if (true === $attributes['abstract']) {
+                $methodEnd = $tokens->getNextTokenOfKind($index, array(';'));
+            } else {
+                $methodStart = $tokens->getNextTokenOfKind($index, array('{'));
+                $methodEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $methodStart, true);
+            }
+
+            $this->fixSpaceBelowMethod($tokens, $classEnd, $methodEnd);
+            $this->fixSpaceAboveMethod($tokens, $classStart, $index);
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $classStart
+     * @param int    $classEnd
+     */
+    private function fixInterface(Tokens $tokens, $classStart, $classEnd)
+    {
+        for ($index = $classEnd; $index > $classStart; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            $methodEnd = $tokens->getNextTokenOfKind($index, array(';'));
+
+            $this->fixSpaceBelowMethod($tokens, $classEnd, $methodEnd);
+            $this->fixSpaceAboveMethod($tokens, $classStart, $index);
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $classEnd
+     * @param int    $methodEnd
+     */
+    private function fixSpaceBelowMethod(Tokens $tokens, $classEnd, $methodEnd)
+    {
+        $nextNotWhite = $tokens->getNextNonWhitespace($methodEnd);
+        $this->correctLineBreaks($tokens, $methodEnd, $nextNotWhite, $nextNotWhite === $classEnd ? 1 : 2);
+    }
+
+    /**
+     * Fix spacing above a method signature. Deal with comments, PHPDocs and spaces above the method.
+     *
+     * @param Tokens $tokens
+     * @param int    $classStart  index of the class Token the method is in
+     * @param int    $methodIndex index of the method to fix
+     */
+    private function fixSpaceAboveMethod(Tokens $tokens, $classStart, $methodIndex)
+    {
+        static $methodAttr = array(T_PRIVATE, T_PROTECTED, T_PUBLIC, T_ABSTRACT, T_FINAL, T_STATIC);
+
+        // find out where the method signature starts
+        $firstMethodAttrIndex = $methodIndex;
+        for ($i = $methodIndex; $i > $classStart; --$i) {
+            $nonWhiteAbove = $tokens->getNonWhitespaceSibling($i, -1);
+            if (null !== $nonWhiteAbove && $tokens[$nonWhiteAbove]->isGivenKind($methodAttr)) {
+                $firstMethodAttrIndex = $nonWhiteAbove;
+            } else {
+                break;
+            }
+        }
+
+        if (false === $tokens[$nonWhiteAbove]->isGivenKind(T_DOC_COMMENT)) {
+            $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstMethodAttrIndex, $nonWhiteAbove === $classStart ? 1 : 2);
+
+            return;
+        }
+
+        // there should be one linebreak between the method signature and the PHPDoc above it
+        $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstMethodAttrIndex, 1);
```

Comment:
does the `NoEmptyLinesAfterPhpdocsFixer` add a line break if it missing (or just removes empty lines as the name suggests)?